specification/dartLangSpec.tex

```diff
@@ -16244,10 +16245,19 @@ \subsection{Identifier Reference}
   where $T$ is a subtype of any other type $S$
   such that $v$ is known to have type $S$,
   in which case the static type of $e$ is $T$.
+\item
+  % Extension getter invocation by scope.
+  If $D$ is a declaration of an instance getter in an extension $E$
+  then the static type of $e$ is the return type of $D$.
+\item
+  % Extension method tear-off by scope.
+  If $D$ is a declaration of an instance method
+  in an extension $E$ then the static type of $e$ is
+  the function type of $D$.
```

Comment:
Do we need to mention type parameters of *E* or are they assumed to be part of *E* (because we say "extension *E*", not "extension declaration *E*")?

Reply:
I don't think so: Given that a lexical lookup has yielded a declaration of a member of an extension declaration $E$, the current location is inside some member of $E$ as well, and this means that the type arguments that occur free in the function type of $D$ are the available denotations of those types (we don't have any actual type arguments that we can use in a substitution, and it's OK to use the type parameters because they are in scope).

I corrected 'extension $E$' to 'extension declaration $E$', because it is actually a declaration.

It is still possible that we will need to perform a generic function instantiation on $e$ (both for a getter invocation and for a method tear-off), but we don't mention this: It would have to be mentioned for _every_ form of expression because they could all occur in a context where a non-generic function type is the context, and they could all have a type which is a generic function type.